hedera-node/src/test/java/com/hedera/services/utils/UnzipUtilityTest.java

```diff
@@ -0,0 +1,74 @@
+package com.hedera.services.utils;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import java.io.File;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class UnzipUtilityTest {
+	@Test
+	public void privateConstructorShallThrowException() {
+		Constructor<?> ctor;
+		try {
+			ctor = UnzipUtility.class.getDeclaredConstructor();
+			ctor.setAccessible(true);
+
+			ctor.newInstance();
+		} catch (Exception e) {
+			IllegalStateException cause = (IllegalStateException) e.getCause();
```

Comment:
nit: here you should catch the specific `IllegalStateException` instead of a generic `Exception`, so you can avoid the unnecessary cast.

Reply:
If doing so, catching `IllegalStateException` alone won't work, we need to also catch all other exception types that can be thrown by `getDeclaredConstructor` and `newInstance`. Essentially it's something like this:

```
		try {
			ctor = UnzipUtility.class.getDeclaredConstructor();
			ctor.setAccessible(true);

			ctor.newInstance();
		} catch (IllegalStateException | NoSuchMethodException | InstantiationException | InvocationTargetException | IllegalAccessException e) {
			IllegalStateException cause = (IllegalStateException) e.getCause();
			assertEquals("UnzipUtility is an utility class. Shouldn't create any instance!", cause.getMessage());
		}

```
A cast is still required. So for a test case, I guess it's probably a little overkill.

Reply:
hmm, in that case the `cast` is not quite right, because what if you really received other type of `Exception`? Then the correct solution should be catching `IllegalStateException` for the assertion then catch other generic `Exception` just to ignore for the passing of the test.

Reply:
Yeah, should be something like above updated.